src/main/java/org/hydev/mcpm/client/database/searchusecase/SearcherByCommand.java

```diff
@@ -0,0 +1,40 @@
+package org.hydev.mcpm.client.database.searchusecase;
+
+import org.hydev.mcpm.client.models.PluginModel;
+
+import java.util.*;
+
+/**
+ * Searcher that returns a map based on commands.
+ *
+ * @author Jerry Zhu (<a href="https://github.com/jerryzhu509">...</a>)
+ */
+public class SearcherByCommand implements Searcher {
+
+    private static Map<String, List<PluginModel>> commandMap = null;
+
+    /**
+     * Returns a dictionary mapping the different commands to the matching plugins.
+     *
+     * @param plugins A list of all plugins in the database.
+     * @return A dictionary associating the command to the matching plugins.
+     */
+    @Override
+    public Map<String, List<PluginModel>> constructSearchMaps(List<PluginModel> plugins) {
```

Comment:
Personally I would consider this an implementation detail, as we can see it already varies, for example the SearcherByKeyword actually uses sets of plugins in the map.
This means it shouldn't be exposed to the public (you can remove this from the interface).

Reply:
i dont follow exactly.  ill leave it for now